src/shared.c

```diff
@@ -364,7 +370,7 @@ gchar **squash_get_matching_files_install_icons_and_mime_data(sqfs* fs, char* pa
                 gchar *dest = NULL;
                 if(inode.base.inode_type == SQUASHFS_REG_TYPE) {
                     if(g_str_has_prefix(trv.path, "usr/share/icons/") || g_str_has_prefix(trv.path, "usr/share/pixmaps/") || (g_str_has_prefix(trv.path, "usr/share/mime/") && g_str_has_suffix(trv.path, ".xml"))){
-                        gchar *path = replace_str(trv.path, "usr/share", g_get_user_data_dir());
+                        gchar *path = replace_str(trv.path, "usr/share", xdg_data_home());
```

Comment:
xdg_data_home() <- probable memory leak

Reply:
Must've missed this call while fixing the leaks everywhere else. Thanks.